src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingAssignmentOperator.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class LocalRewriter
+    {
+        public override BoundNode VisitNullCoalesingAssignmentOperator(BoundNullCoalesingAssignmentOperator node)
+        {
+            SyntaxNode syntax = node.Syntax;
+            var temps = ArrayBuilder<LocalSymbol>.GetInstance();
+            var stores = ArrayBuilder<BoundExpression>.GetInstance();
+
+            // Rewrite LHS with temporaries to prevent double-evaluation of side effects, as we'll need to use it multiple times.
+            BoundExpression transformedLHS = TransformCompoundAssignmentLHS(node.LeftOperand, stores, temps, node.LeftOperand.HasDynamicType());
+            var lhsRead = TransformNullCoalescingLHSRead(transformedLHS, stores, temps);
+            BoundExpression loweredRight = VisitExpression(node.RightOperand);
+
+            // Now that LHS is transformed with temporaries, we rewrite this node into a conditional expression:
+            // (xlhs != null) ? xlhs : (xlhs = rhs)
+
+            // xlhs != null
+            BoundExpression nullCheck = MakeNullCheck(syntax, lhsRead, BinaryOperatorKind.NotEqual);
+
+            // xlhs = rhs
+            BoundExpression assignment = MakeAssignmentOperator(syntax, transformedLHS, loweredRight, node.LeftOperand.Type, used: true, isChecked: false, isCompoundAssignment: true);
```

Comment:
>isChecked: false [](start = 137, length = 16)

The `isChecked: false` feels strange. Shouldn't that depend on the checked context?